### ISSUE TITLE
How to use GPU? / Check that indeed nixos-wsl uses the GPU?

### ISSUE THREAD

**Issue by doronbehar** [nix-community/NixOS-WSL#264]
I tried to run:

```
env NIXPKGS_ALLOW_UNFREE=1 nix run --impure nixpkgs#nvtop
```

And I got:

```
No GPU to monitor.
```

Whereas on my NixOS laptop it successfully detects my intel GPU. Another method I tried is using `glxgears` (from package `glxinfo`). It doesn't fail to run, but it won't open any window...

My GPU is Intel(R) UHD Graphics 750, and I have the following in my `configuration.nix`:

```nix
hardware.opengl = {
  enable = true;
  extraPackages = [
    pkgs.intel-compute-runtime
    pkgs.rocm-opencl-icd
  ];
};
```

**Comment by SuperSandro2000** [comment]
I have wslg turned and working correctly (I can open xyes and glxgears) but nvtop does not find any gpu, too.

**Comment by nzbr** [comment]
![image](https://github.com/nix-community/NixOS-WSL/assets/7851175/63282ed9-0623-4537-8569-4ee1b1d2f00e)

It works for me without any special opengl packages

**Comment by doronbehar** [comment]
> It works for me without any special opengl packages

Are you enabling anything `services.xserver` related in your config? If so, could you share please?

**Comment by nzbr** [comment]
I just tried it with a fresh install from the latest CI run and it works with that as well

**Comment by doronbehar** [comment]
Hey sorry for the late reply. I'm running this nixos-wsl on my office's machine and only now I got the chance to test `xeyes` on it. Unfortunately, it doesn't work:

```
X Error of failed request: BadAlloc (insufficient resources for operation)
  Major opcode of failed request:  53 (X_CreatePixmap)
  Serial number of failed request:  56
  Current serial number in output stream:  67
```

However, I have some python scripts that generate matplotlib plots that _do_ manage to get opened, so indeed this is peculiar.

@nzbr Could you share please from where can I grab a tarball from the CI? I installed mine from the latest release, and then updated it with `nixos-rebuild`. My full configuration is available here:

https://gitlab.com/doronbehar/nixos-configs/-/blob/42a174afaba248d0888dd447b57c17921fbd2533/flake.nix#L106-125

**Comment by nzbr** [comment]
You can use the "installer" artifact from here: https://github.com/nix-community/NixOS-WSL/actions/runs/5325165561
The tarball is inside a zip. You need to unpack that and then you can import the tarball

**Comment by doronbehar** [comment]
Weird. I didn't manage to debug why this didn't work with the release's tarball. I used the tarball from that CI run and now it works, also with my configuration from https://gitlab.com/doronbehar/nixos-configs ...

I guess this can be closed if it'd be considered a "support" thread because I don't have a way to further debug what was the issue (I accidentally `--unregister`ed that installation).

I do suspect that this had something to do with the fact that I changed the main user of the machine in the previous installation - and that didn't go well regarding the IDs in `/mnt/wslg` - also related to our discussion at https://github.com/nix-community/NixOS-WSL/issues/250#issuecomment-1598918396 .

Thanks for your help in any case!
